### ISSUE TITLE
Creating templates from GraphQL data

### ISSUE THREAD

**Issue by christianlemmers** [gridsome/gridsome#870]
## Description

I'm trying to source entries from CraftCMS using the [source-graphql](https://gridsome.org/plugins/@gridsome/source-graphql) plugin and let Gridsome build pages for these entries. Unfortunatelly I can't seem to get the right content type because everytime I'm trying to build the error message says:
`A content type for the categories template does not exist`

#### gridsome.config.js
```
plugins: [
        {
            use: '@gridsome/source-graphql',
            options: {
                url: process.env.CRAFT_API_URL,
                fieldName: 'craft',
                typeName: 'craft',
                headers: {
                    Authorization: `Bearer ${process.env.CRAFT_API_TOKEN}`,
                }
            }
        }
    ],
templates: {
        categories: [
            {
                path: '/category/:id',
                component: './src/templates/Category.vue'
            }
        ]
    },
```

I'm not sure what's going wrong. It seems like the type- and fieldName is conflicting with the templates.

**Comment by Al-Rozhkov** [comment]
Please see https://github.com/gridsome/gridsome/issues/770

This is limitation of `source-graphql` plugin. It doesn't creates collections and nodes, just merges gridsome graphql with remote schema. Especially sad that it is not documented.

**Comment by hjvedvik** [comment]
@Al-Rozhkov Yes, it should probably be mentioned in the readme file for the plugin.

And I understand that it's confusing that it behaves differently than other source plugins. We could maybe rename it to `@gridsome/graphql-remote-schema` or something to make it clearer

**Comment by indirectlylit** [comment]
Super confusing ... I just spent a few hours trying to figure out how to do this, since it seemed from the docs that GraphQL was supposed to be a first-class citizen in Gridsome.

Here are some related questions found in Discord:

![image](https://user-images.githubusercontent.com/2367265/71611475-2fb15c80-2b67-11ea-8a27-7bcc118a5a00.png)

![image](https://user-images.githubusercontent.com/2367265/71611490-5bccdd80-2b67-11ea-86ce-fcff5f517996.png)

![image](https://user-images.githubusercontent.com/2367265/71611506-769f5200-2b67-11ea-9423-8ea21aa7dfc0.png)

**Comment by sanscheese** [comment]
@christianlemmers my plugin will help with the template handling the craft cms GraphQL endpoint with Gridsome: https://www.npmjs.com/package/@bhws/gridsome-source-craft-graphql

I'm also starting to set up my site and write some articles on this, so will have better starting points soon. Current I have this article:
https://bensheedy.com/blog/live-preview-with-craft-cms-and-gridsome

**Comment by MartinN3** [comment]
Strapi could benefit from the usage too i mean instead of importing data from API using gridsome-strapi-plugin.

**Comment by kwiat1990** [comment]
> Please see #770
> 
> This is limitation of `source-graphql` plugin. It doesn't creates collections and nodes, just merges gridsome graphql with remote schema. Especially sad that it is not documented.
> 
> You can use [this solution](https://gridsome.org/docs/pages-api/#create-pages-from-graphql), it's not very elegant though.

Are some other plugins  out there, which can probably manage it to work? I mean it will be nice to have all this things like slug and date out of the box. These doesn't work for pages.
